arch/sim/src/sim/up_i2cbus.c

```diff
@@ -0,0 +1,153 @@
+/****************************************************************************
+ * arch/sim/src/sim/up_i2cbus.c
```

Comment:
don't need up_i2cbus.c, let up_linuxi2cbus.c implement sim_i2cbus_initialize/sim_i2cbus_uninitialize directly.

Reply:
The reason I put this here was I figured that if another I2C bus implementor was here like libhid that this would resolve the mapping.

I can move it out it just means only supporting one at a time.

Reply:
Actually, the implementation isn't much difference between an i2c hardware driver and an i2c linux/macOS wrapper driver. The key point is that i2c linux/macOS wrapper driver can't include <nuttx/i2c/i2c_master.h> directly, so the natural solution is:

1. Create a header file contain required i2c macro/callback/strcuture
2. Let up_i2cbuslinux.c or up_i2cbusmacos.c include this header file and implement the callback
3. Implement sim_i2c_initialize and sim_i2c_uninitialize
4. Add Kconfig option to select the source file to be built 

In step 2, we normally include the host specific header file(e.g. linux/i2c-dev.h) and map NuttX API to host API. Since you can do what ever you want in up_i2cbuslinux.c or up_i2cbusmacos.c, it's redundant to add another layer(e.g. up_i2cbus.c).

Reply:
Ok that's fine. I guess if we were to support using other userspace I2C libraries (via libftdi, libhid, etc) they could just be mapped and initialized as appropriate in the host specific bus code.

Reply:
Done.